hadoop-ozone/fault-injection-test/mini-chaos-tests/src/test/java/org/apache/hadoop/ozone/MiniOzoneLoadGenerator.java

```diff
@@ -47,212 +42,61 @@
-    LOG.info("AGED LOADGEN: Started Aged IO Thread:{}.", threadID);
-    String threadName = Thread.currentThread().getName();
-    long startTime = Time.monotonicNow();
-
-    while (isIOThreadRunning.get() &&
-        (Time.monotonicNow() < startTime + runTimeMillis)) {
-
-      String keyName = null;
-      try {
-        if (agedWriteProbability.isTrue()) {
-          int index = agedFileWrittenIndex.getAndIncrement();
-          ByteBuffer buffer = getBuffer(index);
-          keyName = getKeyName(index, threadName);
-
-          agedLoadBucket.writeKey(buffer, keyName);
-        } else {
-          Optional<Integer> index = randomKeyToRead();
-          if (index.isPresent()) {
-            ByteBuffer buffer = getBuffer(index.get());
-            keyName = getKeyName(index.get(), threadName);
-            agedLoadBucket.readKey(buffer, keyName);
-          }
-        }
-      } catch (Throwable t) {
-        LOG.error("AGED LOADGEN: {} Exiting due to exception", keyName, t);
-        break;
-      }
-    }
-    // This will terminate other threads too.
-    isIOThreadRunning.set(false);
-    LOG.info("Terminating IO thread:{}.", threadID);
-  }
-
-  // Start IO load on an Ozone bucket.
-  private void startFsLoad(long runTimeMillis) {
-    long threadID = Thread.currentThread().getId();
-    LOG.info("Started Filesystem IO Thread:{}.", threadID);
-    String threadName = Thread.currentThread().getName();
-    long startTime = Time.monotonicNow();
-
-    while (isIOThreadRunning.get() &&
-      (Time.monotonicNow() < startTime + runTimeMillis)) {
-      try {
-        int index = RandomUtils.nextInt();
-        ByteBuffer buffer = getBuffer(index);
-        String keyName = getKeyName(index, threadName);
-        fsBucket.writeKey(true, buffer, keyName);
-
-        fsBucket.readKey(true, buffer, keyName);
-
-        fsBucket.deleteKey(true, keyName);
-      } catch (Exception e) {
-        LOG.error("LOADGEN: Exiting due to exception", e);
-        break;
-      }
+  private final List<LoadExecutors> loadExecutors;
+
+  MiniOzoneLoadGenerator(OzoneVolume volume, int numClients, int numThreads,
+                         int numBuffers, OzoneConfiguration conf)
+      throws Exception {
+    DataBuffer buffer = new DataBuffer(numBuffers);
+    loadExecutors = new ArrayList<>();
+
+    // Random Load
+    String mixBucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+    volume.createBucket(mixBucketName);
+    List<LoadBucket> ozoneBuckets = new ArrayList<>(numClients);
+    for (int i = 0; i < numClients; i++) {
+      ozoneBuckets.add(new LoadBucket(volume.getBucket(mixBucketName),
+          conf));
     }
-    // This will terminate other threads too.
-    isIOThreadRunning.set(false);
-    LOG.info("Terminating IO thread:{}.", threadID);
+    RandomLoadGenerator loadGenerator =
+        new RandomLoadGenerator(buffer, ozoneBuckets);
+    loadExecutors.add(new LoadExecutors(numThreads, loadGenerator));
+
+    // Aged Load
```

Comment:
NIT: line 65-84 can be refactored to dedup the same create loadgenerator and added to load executors.

Reply:
done